portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayFormInput.testcase

```diff
@@ -0,0 +1,180 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Clay";
+	property testray.main.component.name = "User Interface";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+        JSONLayout.addPublicLayout(
+			groupName = "Guest",
+			layoutName = "Clay Sample Test Page");
+
+        JSONLayout.updateLayoutTemplateOfPublicLayout(
+			groupName = "Guest",
+			layoutName = "Clay Sample Test Page",
+			layoutTemplate = "1 Column");
+
+        Navigator.gotoPage(pageName = "Clay Sample Test Page");
+
+		Portlet.addPG(portletName = "Clay Sample");
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+        else {
+			Page.tearDownCP();
+		}
+	}
+
+	@priority = "4"
+	test ClayCheckbox {
+        Navigator.gotoNavTab(navTab = "Form Elements");
+
+        AssertChecked.assertCheckedNotVisible(
+            key_definition = "On",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertNotChecked.assertNotCheckedNotVisible(
+            key_definition = "Off",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertEditable.assertEditableNotVisible(
+            locator1 = "ClaySamplePortlet#CHECKBOX_WITH_CUSTOM_CLASS_AND_ID");
+
+        Check.checkNotVisible(
+            checkboxName = "Checkbox with Label",
+            locator1 = "Checkbox#ANY_CHECKBOX");
+
+        AssertNotEditable.assertNotEditableNotVisible(
+            key_definition = "On disabled",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertChecked.assertCheckedNotVisible(
+            key_definition = "On disabled",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertNotEditable.assertNotEditableNotVisible(
+            key_definition = "Off disabled",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertNotChecked.assertNotCheckedNotVisible(
+            key_definition = "Off disabled",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        AssertEditable.assertEditableNotVisible(
+            key_definition = "Indeterminate",
+            locator1 = "ClaySamplePortlet#CHECKBOX");
+
+        SikuliAssertElementPresent(locator1 = "ClaySamplePortlet#CHECKBOX_INDETERMINATE_PNG");
```

Comment:
oh it looks like the proper way to test may be within a nested checkbox: https://css-tricks.com/indeterminate-checkboxes/. So this looks valid.

Reply:
I used SikuliAssertElementPresent in this test to assert the indeterminate checkbox, this is the only way I can think of, but it's a little difficult to make the sikuli images work, when I try on Windows with the images that I took on the same laptop, this function was not quite stable. Do you have some ideas that if we can assert with some other methods?

Reply:
yea...I would like to avoid using Sikuli as much as possible. I plan to research more about other alternatives we can bring into poshi/ci. @calvinctvo what was that visual testing program you worked with before with flex? Do you think that would be more stable in regression testing icons/visual differences?

Reply:
@calvinctvo mentioned TestCafe which uses Percy. I think we'd like some sort of stable baseline visual regression tool like that. I added it for something to research later: https://issues.liferay.com/browse/LPS-124435. 

For now, I think AssertChecked and AssertNotChecked is ok to verify the states and we'll accept the risk on the horizontal visual until we get more stable visual testing for that usecase.

Reply:
OK, thanks for creating the ticket!